### ISSUE TITLE
Improving PyScript docs

### ISSUE THREAD

**Issue by marimeireles** [pyscript/pyscript#579]
This is an umbrella issue to coordinate documentation efforts.

When adding new pieces of documentation keep in mind we're trying to have the (diataxis)[https://diataxis.fr/] system. If you're not into text maybe check this [PyCon's talk](https://www.youtube.com/watch?v=t4vKPhjcMZg ).

- [x] #580
- [x] #581
- [x] #582
- [x] #584
- [x] #583
- [ ] https://github.com/pyscript/pyscript/issues/756
- [ ] Write docs for how to contribute to the new test system https://github.com/pyscript/pyscript/issues/601

Please feel free to add ideas either by editing this comment or adding your own comments underneath.

**Comment by ahrnbom** [comment]
I don't know if this is the right place to comment about this, but I recently learned about PyScript and is very intrigued by the idea. The first thing I wanted to know was how to make an HTTP call to the server that's hosting the website with PyScript files. I spent about an hour looking through the documentation, but couldn't find anything more than a vague description that all the Web APIs were available. But that linked to Mozilla's JS APIs, not PyScript's Python APIs. This confused me a lot. I would assume I am not alone in this.

It took me until I stumbled upon this sentence in the seemingly unrelated documentation page about the DOM until I understood: 

```
It means the API defined in JavaScript remains the same in Python, so any browser based JavaScript APIs or third party JavaScript libraries that expose objects in the web page's globalThis, will have exactly the same API in Python as in JavaScript.
```

To me, this is very cryptic. Explicit is better than implicit. So my suggestion would be to actually create proper documentation for all the APIs that you have implemented, along with the full descriptions of which Python types to use where, which calls are async or synchronous, etc. The current approach requires a bit of guess work from the programmer. Unless I am missing something, of course.

Thanks for making this awesome tool :)
